examples/advanced/quantum_volume.py

```diff
@@ -151,19 +247,36 @@ def main(num_qubits: int, depth: int, num_repetitions: int, seed: int):
         num_qubits: The number of qubits for the circuit.
         depth: The number of gate layers to generate.
         num_repetitions: The number of times to run the algorithm.
+        samplers: The samplers to run the algorithm on.
+        deconstruct: An optional function to deconstruct the model circuit's
```

Comment:
hmm

1. I think the name `deconstruct` is unexpected. Maybe "compiler"?
2. What's the difference between this and `optimize`? 
3. You could consider also taking a callable (or similar) to control which routing algorithm is used

Reply:
1. Changed to `compiler`.
2. The difference is that the `deconstruct` (now `compiler`) step doesn't do any optimizations (e.g. combining of gates) - instead, it just decomposes the model gates down into a gate set supported by the device. `optimize` actually elides and combines gates after routing is complete.
3. Done.